### PR TITLE
[CELEBORN-883][WORKER] Optimized configuration checks during MemoryManager initialization

### PR DESCRIPTION
**Pull request by zwangsheng**
### What changes were proposed in this pull request?
1. Expose the config check logic during `MemoryManager#initialization` in the user configuration doc.
2. Add Preconditions Error Message
3. Add unit test to make sure that part of the logic isn't altered by mistake

### Why are the changes needed?
User-friendly

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
Add Unit Test

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2450,7 +2450,8 @@ object CelebornConf extends Logging {
   val WORKER_DIRECT_MEMORY_RATIO_PAUSE_REPLICATE: ConfigEntry[Double] =
     buildConf("celeborn.worker.directMemoryRatioToPauseReplicate")
       .categories("worker")
-      .doc("If direct memory usage reaches this limit, the worker will stop to receive replication data from other workers.")
+      .doc("If direct memory usage reaches this limit, the worker will stop to receive replication data from other workers." +
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2453]:
```suggestion
      .doc("If direct memory usage reaches this limit, the worker will stop to receive replication data from other workers. " +
```

File: worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java [hunk 2]
```diff
@@ -120,7 +120,12 @@ private MemoryManager(CelebornConf conf) {
             .<Long>invoke();
 
     Preconditions.checkArgument(maxDirectorMemory > 0);
-    Preconditions.checkArgument(pauseReplicateRatio > pausePushDataRatio);
+    Preconditions.checkArgument(
+        pauseReplicateRatio > pausePushDataRatio,
+        String.format(
+            "Invalid config, {} should be greater than {}",
```
Comment (pan3793) [inline worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:126]: let's also mention the value in the error message
```suggestion
            "Invalid config, {}({}) should be greater than {}({})",
```

File: worker/src/test/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManagerSuiteJ.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+package org.apache.celeborn.service.deploy.worker.memory;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.celeborn.common.CelebornConf;
+
+public class MemoryManagerSuiteJ {
```
Comment (pan3793) [inline worker/src/test/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManagerSuiteJ.java:26]: scala is preferred for brand new testing

Comment (zwangsheng) [reply worker/src/test/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManagerSuiteJ.java:26]: Good Idea!

**Review by pan3793 — APPROVED** [review]

**Comment by zwangsheng** [comment]
thanks, merge to main

**Pull request closed without merging**